### DISCUSSION TITLE
PDE with a spatially correlated stochastic term

### DISCUSSION THREAD

**Discussion by JonasVeenstra** [zwicker-group/py-pde#613] [Q&A] [answered]
Hi Dr. Zwicker,

Thanks for developing a great and userfriendly package!

I understand the package supports PDE's with additive noise but so far I've only been able to generate a Gaussian noise term that depends both on time and space, as in the following PDE:

![image](https://github.com/user-attachments/assets/08359c61-de52-4d3d-9387-05da432942a0)

However, I would like the noise to be white only in time and harmonic in space, similar to the classmethod [random_harmonic](https://py-pde.readthedocs.io/en/latest/_modules/pde/fields/datafield_base.html#DataFieldBase.random_harmonic):

![image](https://github.com/user-attachments/assets/9c7ddd97-0acf-4d8d-9891-d19f6d88ef4a)

with a the amplitude, L the box size and j the mode number.

That mentioned classmethod seems intended to be used for generating a field as an initial condition, but I would like it to function as the noise term. Is there a way to do this? If so, could you point me in the right direction?

Many thanks in advance!

**Comment by david-zwicker** [comment, marked as answer]
You can define your own noise term by subclassing `PDEBase`; see [documentation](https://py-pde.readthedocs.io/en/latest/packages/pde.pdes.base.html#pde.pdes.base.PDEBase). Essentially, you need to overwrite the methods `noise_realization` (for the numpy backend) and/or `_make_noise_realization_numba` (for the numba backend).

**Reply by JonasVeenstra** [reply]
Many thanks, I got it to work! (I overlooked this [example](https://py-pde.readthedocs.io/en/latest/examples_gallery/custom_noise.html#custom-noise) which helped a lot.)

**Reply by david-zwicker** [reply]
It's not surprising that you didn't see the example, since I just added it inspired by your question :)
